### PR TITLE
Migrate further areas to room

### PR DESCRIPTION
**Pull request by AndyScherzinger**
* settings
* profile
* switching accounts

I keep synchronous process synchronous (i.e. always calling blockingGet) because the current focus is to get rid of any requery interactions with the database especially since the controllers aren't here to be kept nor is their individual implementation build for async atm.

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@mahibi I couldn't reproduce the account switching issue from #2218 with my latest changes on this branch maybe because this update to room now ships with a set active user method that wraps all changes into a single transaction.

**Comment by mahibi** [comment]
> @mahibi I couldn't reproduce the account switching issue from #2218 with my latest changes on this branch maybe because this update to room now ships with a set active user method that wraps all changes into a single transaction.

this doesn't fix the issue, reproduced it with this branch :(

**Comment by AndyScherzinger** [comment]
> this doesn't fix the issue, reproduced it with this branch :(

😢 then I was just lucky to not have hit the issue I guess. 
Anyway any review and 👍 is highly appreciated, so I can work on the follow-up PRs more easily :)

**Comment by mahibi** [comment]
> > this doesn't fix the issue, reproduced it with this branch :(
> 
> cry then I was just lucky to not have hit the issue I guess. Anyway any review and +1 is highly appreciated, so I can work on the follow-up PRs more easily :)

i will take another look regarding the issue. i think the issue appeared when writing a message but the conversation list before was correct if i remember correctly. So there might just be more parts that need the `.blockingGet()` in order to fix ALL scenarios..?

**Comment by AndyScherzinger** [comment]
Thanks. I can't say for sure, since the behavior we have seen was rather strange because it seems that some async operation would mix-up the user-context which wasn't on DB level since restarting the app showed the correct data, just any further calls went wrong which makes me believe it is either something in a worker happening in the background or some cache issue in a library like okhttp

**Review by mahibi — APPROVED** [review]

File: app/src/main/java/com/nextcloud/talk/users/UserManager.kt [hunk 1]
```diff
@@ -123,6 +123,18 @@ class UserManager internal constructor(private val userRepository: UsersReposito
         }.toSingle()
     }
 
+    fun updateUser(user: User): Single<Int> {
+        return if (user.id != null) {
+            Single.just(userRepository.updateUser(user))
+        } else {
+            Single.just(userRepository.insertUser(user).toInt())
+        }
```
Comment (AlvaroBrey) [inline app/src/main/java/com/nextcloud/talk/users/UserManager.kt:131]:
```suggestion
        return Single.fromCallable {
            when (user.id) {
                null -> userRepository.insertUser(user).toInt()
                else -> userRepository.updateUser(user)
            }
        }
```
Simplified code a bit.

I'd prefer if we defined a `Single` version of `updateUser/insertUser` and just return a call to that, but this is fine for now.

**Review by AlvaroBrey — APPROVED** [review]

**Pull request merged**